src/components/Toggle/Toggle.tsx

```diff
@@ -23,9 +24,10 @@ type Props = {
    * @default false
    */
   disabled?: boolean;
-} & Omit<InputHTMLAttributes<HTMLInputElement>, 'children' | 'onChange'>;
+} & Omit<InputHTMLAttributes<HTMLInputElement>, 'children' | 'onChange'> &
+  CustomDataAttributeProps;
 
-export const Toggle: FC<Props> = forwardRef<HTMLLabelElement, Props>(
+export const Toggle: FC<Props> = forwardRef<HTMLInputElement, Props>(
```

Comment:
The actual state of the Toggle is input, so it can take input refs.
The fact that it is Props with an input assumption is also a factor in the change.